avldb/src/main/scala/scorex/db/LDBKVStore.scala

```diff
@@ -45,9 +52,18 @@ class LDBKVStore(protected val db: DB) extends KVStoreReader with ScorexLogging
     }
   }
 
-  def insert(values: Array[(K, V)]): Try[Unit] = update(values, Array.empty)
+  /**
+    * `update` variant where we only insert values into this database
+    */
+  def insert(keys: Array[K], values: Array[V]): Try[Unit] = update(keys, values, Array.empty)
+
+  def insert(values: Array[(K, V)]): Try[Unit] = update(values.map(_._1), values.map(_._2), Array.empty)
```

Comment:
This code is used for all database inserts, doesn't .map slow it down?

Reply:
Yes, on this system level better to avoid such accidental overhead. Ideally the `values: Array[(K, V)]` should never be created, because it allocates each (K, V) pair individually, only to be deconstructed back in this method. 
One way to avoid this is to use Coll[(K, V)] from Sigma. It keeps Ks and Vs in two separate arrays so zip and unzip are O(1) operation.

Reply:
eliminated most of calls creating paired argument in https://github.com/ergoplatform/ergo/pull/1992 , as well as insert with paired argument